src/realm/object-store/sync/sync_session.cpp

```diff
@@ -633,21 +634,24 @@ void SyncSession::handle_error(sync::SessionErrorInfo error)
 {
     enum class NextStateAfterError { none, inactive, error };
     auto next_state = error.is_fatal ? NextStateAfterError::error : NextStateAfterError::none;
-    auto error_code = error.status.get_std_error_code();
     util::Optional<ShouldBackup> delete_file;
     bool log_out_user = false;
     bool unrecognized_by_client = false;
 
+    // Can be called from within SyncSession with an AuthError to handle app-level authentication problems,
+    // so even if there's an error action, we need to set a fall-back to logging out the user and making
```

Comment:
when there is no error action (i.e, a synthetized error), we'll set `unrecognized_by_client` to true. I don't think we want that. This behavior is different than before.